src/CodeGen.fs

```diff
@@ -1175,7 +1204,11 @@ type GraphqlErrorResponse = {{ errors: {errorType} list }}
 
 type {clientName}(url: string, httpClient: HttpClient, options: JsonSerializerOptions) =
     new(url: string, options: JsonSerializerOptions) = {clientName}(url, new HttpClient(), options)
+    new(url: string, httpClient: HttpClient) = {clientName}(url, httpClient, new JsonSerializerOptions())
     new(url: string) = {clientName}(url, new HttpClient(), new JsonSerializerOptions())
+    new(httpClient: HttpClient, options: JsonSerializerOptions) = {clientName}(String.Empty, httpClient, options)
+    new(httpClient: HttpClient) = {clientName}(String.Empty, httpClient, new JsonSerializerOptions())
```

Comment:
Using `String.Empty` will break the runtime implementation, right? Because the `url` parameter is still used when making the `POST` request. When using this constructor, I assume users have provided the URL to the GraphQL service via the `BaseAddress` of the `HttpClient`. Can you please use that for the `url`? 
> You probably have to check first that `httpClient.BaseAddress` is not null and then use `httpClient.BaseAddress.OriginalString` for the `url` parameter. If it is `null`, then we throw a null argument exception saying that the HttpClient is missing a base address

cc @xperiandri

Reply:
@dim-37, yes, we need to check that `BaseAddress` is set and check what value must be provided as URL, will it be `null` or `String.Empty`